Header header
string recipient
uint8 state